neuland/calibration/calLevel/R3BFTCalEngine.cxx

```diff
@@ -0,0 +1,178 @@
+#include "R3BFTCalEngine.h"
+#include "R3BException.h"
+#include <range/v3/all.hpp>
+
+namespace R3B::Neuland::calibration
+{
+    auto FTType2Str(FTType type) -> std::string_view
+    {
+        switch (type)
+        {
+            case FTType::leftleading:
+                return "left_leading";
+            case FTType::rightleading:
+                return "right_leading";
+            case FTType::lefttrailing:
+                return "left_trailing";
+            case FTType::righttrailing:
+                return "right_trailing";
+            case FTType::trigger:
+                return "trigger";
+            default:
+                throw R3B::logic_error("unresolvable types");
+        }
+    }
+
+    ModuleCal::ModuleCal(std::string_view hist_name, unsigned int mID)
+        : FTBaseCal{ hist_name, mID, { FTType::trigger } }
+    {
+        auto initHist = R3B::make_hist<TH1I>(hist_name.data(), hist_name.data(), MaxFTValue, 0.5, MaxFTValue + 0.5);
+        InitAllDistributions(initHist.get());
+    }
+
+    PlaneCal::PlaneCal(std::string_view hist_name, unsigned int mID)
+        : FTBaseCal{ hist_name,
+                     mID,
+                     { FTType::leftleading, FTType::lefttrailing, FTType::rightleading, FTType::righttrailing } }
+    {
+        auto initHist = R3B::make_hist<TH2I>(hist_name.data(),
+                                             hist_name.data(),
+                                             BarsPerPlane,
+                                             +0.5,
+                                             BarsPerPlane + 0.5,
+                                             MaxFTValue,
+                                             +0.5,
+                                             MaxFTValue + 0.5);
+        InitAllDistributions(initHist.get());
+    }
+
+    namespace
+    {
+        using ValueErrors = FTCalStrategy::ValueErrors;
+        constexpr unsigned int uniform_err_divider = 12;
+        constexpr double uniform_err_divider_sqrt = 3.464101615137755;
+        constexpr double sqrt_3 = 1.732050807568877;
```

Comment:
Why do you initialize that constant with a fixed value?
Is it because std::sqrt is not constexpr (until C++26?) and you want to save the extra calculation at runtime?
I do not think that two sqrt calls per run are going to affect runtime that much. Some readers like me will have limited math ability and not be able to verify in their head that 3.464101615137755 is in fact the double constant best approximating sqrt(12). This makes it harder to verify that the code does what it claims to do.

Reply:
Yes, exactly. std::sqrt is not constexpr and sqrt function is actually slow compared to other math functions. The thing is the calling time of sqrt is not two calls per run but is equal to the "number of digitizers" times "number of channels", which is around 2600 x 4 x 500. That's a lot. So I think maybe just make it as a constexpr value without calling sqrt. And for now the calibration is done in the end of the run. Maybe in the future we could more calibrations in a single run.

>Some readers like me will have limited math ability and not be able to verify in their head that 3.464101615137755 is in fact the double constant best approximating sqrt(12).

That's true. Naming here is not very intuitive. It could just be `SQRT_12`. But I thought the reason why here is the square root of 12 is because of standard deviation from uniform distribution. Now I think `SQRT_12` is easier to understand.

Reply:
Ah, I guess I will change back to constexpr. If I use `const auto sqrt_12 = std::sqrt(12)`, I have to change all const variables that use sqrt of 12. Actually someone already implement it long time ago:
https://github.com/R3BRootGroup/R3BRoot/blob/b0f85c947428993bf7391c69580463500ac7cd43/neuland/shared/R3BNeulandCommon.h#L25

I will probably use this value instead (and probably change it the name to `SQRT_12` since double underscore prefix is reserved for C++ internal global variable).